src/Dfc.CourseDirectory.Web/Helpers/OnspdSearchHelper.cs

```diff
@@ -1,28 +1,30 @@
-﻿using Dfc.CourseDirectory.Models.Models.Onspd;
-using Dfc.CourseDirectory.Services.Interfaces.OnspdService;
-using Dfc.CourseDirectory.Services.OnspdService;
+﻿using System.Linq;
+using Dfc.CourseDirectory.Core.Search;
+using Dfc.CourseDirectory.Models.Models.Onspd;
+using Mapster;
 
 namespace Dfc.CourseDirectory.Web.Helpers
 {
     public class OnspdSearchHelper : IOnspdSearchHelper
     {
-        private readonly IOnspdService _onspdService;
+        private readonly ISearchClient<Core.Search.Models.Onspd> _searchClient;
 
-        public OnspdSearchHelper(IOnspdService onspdService)
+        public OnspdSearchHelper(ISearchClient<Core.Search.Models.Onspd> searchClient)
         {
-            _onspdService = onspdService;
+            _searchClient = searchClient;
         }
 
         public Onspd GetOnsPostcodeData(string postcode)
         {
             var onspd = new Onspd();
             if (!string.IsNullOrWhiteSpace(postcode))
             {
-                var onspdSearchCriteria = new OnspdSearchCriteria(postcode);
-                var onspdResult = _onspdService.GetOnspdData(onspdSearchCriteria);
-                if (onspdResult.IsSuccess && onspdResult.HasValue)
+                var searchResult = _searchClient.Search(new OnspdSearchQuery() { Postcode = postcode })
+                    .GetAwaiter().GetResult();
+
+                if (searchResult.Results.Count > 0)
                 {
-                    onspd = onspdResult.Value.Value;
+                    return searchResult.Results.Single().Adapt<Onspd>();
```

Comment:
I don't trust AzureSearch to only return one record. Perhaps safer to use `First()`?

Although this could be expressed more easily with `return searchResults.Results.FirstOrDefault()?.Adapt<Onspd>() ?? new Onspd()`? (Would remove nesting and the `onspd` var)

Reply:
The query has a `Size` == `1`